### PR TITLE
fix(tracing): fix span name high cardinality

### PR DESCRIPTION
**Pull request by StarlightIbuki**
### Summary

Service, operation name, and then resource are the three levels of grouping in DataDog.

The operation name(also called span name) is used to filter Datadog APM/service board. High cardinality makes it unusable.

### Checklist

- [x] The Pull Request has tests
- [x] There's an entry in the CHANGELOG
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

Fix FTI-4882

### CODE REVIEW

File: kong/pdk/tracing.lua [hunk 1]
```diff
@@ -55,6 +55,8 @@ local SPAN_KIND = {
   CONSUMER = 5,
 }
 
+local EMPTY = {}
```
Comment (sabertobihwy) [inline kong/pdk/tracing.lua:58]: here `EMPTY` is a module level var, will it pass dirty data from the previous request to the next request?

**Comment by hbagdi** [comment]
Please evaluate this change from a breaking change/behavior perspective.

File: kong/pdk/tracing.lua (hunk 1 above)
Comment (StarlightIbuki) [reply kong/pdk/tracing.lua:58]: It will never be modified, thus it's safe to share the same table.

**Comment by StarlightIbuki** [comment]
> Please evaluate this change from a breaking change/behavior perspective.

It IS a breaking change. I'm handing it over to @kikito so the decision is on him now.

File: kong/tracing/instrumentation.lua [hunk 2]
```diff
@@ -201,7 +211,8 @@ function _M.request(ctx)
     http_flavor = string.format("%.1f", http_flavor)
   end
 
-  local active_span = tracer.start_span(span_name, {
+  local active_span = tracer.start_span("API_gateway", {
```
Comment (samugi) [inline kong/tracing/instrumentation.lua:214]: am I right that this change means otel will lose the "path" information?
Would it make sense to pass `method` and `path` as `attributes` instead so they could be then used by DD to make the `resource` field, while otel wouldn't lose the information?

File: kong/tracing/instrumentation.lua [hunk 3]
```diff
@@ -201,7 +211,8 @@ function _M.request(ctx)
...
+    resource = method .. " " .. path,
```
[the remark this answers is not in the document]
Comment (StarlightIbuki) [reply kong/tracing/instrumentation.lua:215]: I've considered this idea. However, that will be introducing exceptions for resource handling.
For example, in 1 trace we generate, there will only be 1 span that is an HTTP request span, and the definitions of resources for other spans are not clear and depend on what we're tracing.

But it's a good idea to put the `http.target` into the attributes.

[the remark this answers is not in the document]
Comment (StarlightIbuki) [reply kong/tracing/instrumentation.lua:215]: And I also lack knowledge of handling differences between those tracing standards.
Maybe we could refer to how Datadog agents collect other formats of traces?

**Comment by StarlightIbuki** [comment]
Closing because we are rewriting the implementation.

**Pull request closed without merging**